Flaky: access-control test compares two independently-clocked adminClaims() expiries

## What happens

`test/access-control.test.ts > revocation retention > loads the legacy revoked array shape and keeps those revocations biting` fails intermittently:

```
AssertionError: expected 1785838912 to be 1785838913 // Object.is equality
```

The two numbers are Unix timestamps one second apart.

## Mechanism

`adminClaims()` recomputes the clock on **every call**:

```ts
function adminClaims(room: string, jti = 'admin-jti'): RoomTokenClaims {
  const now = Math.floor(Date.now() / 1000);
  return { room, role: 'admin', iat: now, exp: now + 3600, jti };
}
```

and the test calls it twice, expecting the two `exp` values to be equal:

```ts
await ac.serverOptions.revokeEndpoint!.recordRevocation('new-jti', 'm/old', adminClaims('m/old').exp);
await ac.flush();
const state = readState(dir);
expect(state.revoked!['new-jti']).toBe(adminClaims('m/old').exp);
```

Between the two calls the test performs a debounced persist and a `flush()`. If the second boundary of the wall clock ticks anywhere in that window — roughly a 1-in-N chance per run, where N depends on how long the flush takes — the second `adminClaims()` returns an `exp` one greater than the recorded one, and the assertion fails.

## Impact

It reddens whichever PR happens to be running when the boundary is crossed, with a failure in a package that PR did not touch. Observed on #2076 (an MSPDI import change) in run 30895532001. Because the failing job is `Node tests`, it also fails the aggregate `Build + WASM + Rust + Node` gate, so a single unlucky second produces two red checks on an unrelated PR.

The cost is not the red itself but what it teaches: a contributor who sees `collab-server` fail on a viewer PR learns to discount red checks, which is the habit that lets a real failure through.

## Fix

Capture the claims once and reuse them:

```ts
const claims = adminClaims('m/old');
await ac.serverOptions.revokeEndpoint!.recordRevocation('new-jti', 'm/old', claims.exp);
await ac.flush();
const state = readState(dir);
expect(state.revoked!['new-jti']).toBe(claims.exp);
```

Worth grepping the file for the same shape — any other assertion comparing a value derived from one `adminClaims()` call against a fresh one has the identical race.

--- a/packages/collab-server/test/access-control.test.ts
+++ b/packages/collab-server/test/access-control.test.ts
@@ -385,11 +385,14 @@ describe('revocation retention', () => {
       ),
     ).toBeNull();
     // Re-persisting upgrades to the jti -> exp shape with a fallback horizon.
-    await ac.serverOptions.revokeEndpoint!.recordRevocation('new-jti', 'm/old', adminClaims('m/old').exp);
+    // Capture claims once: two independent adminClaims() calls each recompute
+    // Date.now() and can straddle a wall-clock second boundary (#2094).
+    const claims = adminClaims('m/old');
+    await ac.serverOptions.revokeEndpoint!.recordRevocation('new-jti', 'm/old', claims.exp);
     await ac.flush();
     const state = readState(dir);
     expect(typeof state.revoked!['legacy-jti']).toBe('number');
-    expect(state.revoked!['new-jti']).toBe(adminClaims('m/old').exp);
+    expect(state.revoked!['new-jti']).toBe(claims.exp);
   });
 
   it('prunes revocations once their tokens have expired on their own', async () => {

